### PR TITLE
Make the date returned by snapshot script friendly

### LINKED ISSUE #3568: Prometheus 2.0 Feature Request - Snapshot names without colons
Colon in a filename trips up linux tools like ssh, scp, and tar ([shockingly](http://steve-parker.org/linux/tar-colon-filename/)). As a result, writing the script to archive, copy between systems, and restore a snapshot took 3x longer than it had to. I'd propose changing the name format for snapshots to be more script-friendly

(moved from https://github.com/prometheus/tsdb/issues/221)

### PR DESCRIPTION
**Pull request by Gouthamve**
Fixes #3568

Not sure if the right approach. @fabxc @brian-brazil

---
This change is [Reviewable](https://reviewable.io/reviews/prometheus/prometheus/3570)

### CODE REVIEW

**Comment by Gouthamve** [comment]
@SuperQ @robfig

**Comment by brian-brazil** [comment]
I'd use hyphens to avoid having both hyphens and underscores.

File: docs/querying/api.md [hunk 1]
```diff
@@ -389,7 +389,7 @@ These are APIs that expose database functionalities for the advanced user. These
 We also expose a gRPC API whose definition can be found [here](https://github.com/prometheus/prometheus/blob/master/prompb/rpc.proto). This is experimental and might change in the future.
 
 ### Snapshot
-Snapshot creates a snapshot of all current data into `snapshots/<datetime>-<rand>` under the TSDB's data directory and returns the directory as response.
+Snapshot creates a snapshot of all current data into `snapshots/<datetime>-<rand>` under the TSDB's data directory and returns the directory as response. We replace the `:` in the `datetime` with `_` to be script friendly.
```
Comment (brian-brazil) [inline docs/querying/api.md]: We never specify the format of the date time, so there's no need to mention the change.

**Comment by Gouthamve** [comment]
@brian-brazil Yes, but if you want to convert it back to RFC3339 in some python-script, I think its easier to just replace the `_` with `:`.

**Comment by SuperQ** [comment]
[ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) supports a format with no special characters, and is also script friendly. ie.`20171210T201116Z`

**Comment by Gouthamve** [comment]
Done

**Pull request merged**